packages/contentstack-auth/src/utils/totp-handler.ts

```diff
@@ -0,0 +1,135 @@
+import { cliux, configHandler, NodeCrypto, log } from '@contentstack/cli-utilities';
+import { authenticator } from 'otplib';
+import { askOTP } from './interactive';
+
+/**
+ * @class
+ * TOTP handler for managing TOTP-based two-factor authentication
+ */
+class TOTPHandler {
+  private readonly encrypter: NodeCrypto;
+
+  constructor() {
+    this.encrypter = new NodeCrypto();
+  }
+
+  /**
+   * Validates if a string is a valid base32 secret
+   * @param secret The secret to validate
+   * @returns true if valid, false otherwise
+   */
+  private isValidBase32(secret: string): boolean {
+    // Base32 string must:
+    // 1. Contain only uppercase letters A-Z and digits 2-7
+    // 2. Be at least 16 characters long (before padding)
+    // 3. Have valid padding (no single = character)
+    const base32Regex = /^[A-Z2-7]+(?:={2,6})?$/;
+    const nonPaddedLength = secret.replace(/=+$/, '').length;
+    return base32Regex.test(secret) && nonPaddedLength >= 16;
+  }
+
+  /**
+   * Generates a TOTP code from a provided secret
+   * @param secret The TOTP secret to use
+   * @returns string The generated TOTP code
+   * @throws Error if the secret is invalid or code generation fails
+   */
+  generateTOTPFromSecret(secret: string): string {
+    log.debug('Generating TOTP code from provided secret', { module: 'totp-handler' });
+    
+    try {
+      // Validate and normalize secret
+      const normalizedSecret = secret.toUpperCase();
+      if (!this.isValidBase32(normalizedSecret)) {
+        throw new Error('Something went wrong with your authentication setup. Please check the secret or try again');
+      }
+
+      // Generate TOTP code
+      const code = authenticator.generate(normalizedSecret);
+      log.debug('Generated TOTP code successfully', { module: 'totp-handler' });
+      return code;
+    } catch (error) {
+      log.debug('Failed to generate TOTP code', { module: 'totp-handler', error });
```

Comment:
here error object is missed for the info level, it has to logged in error  level, or re throw the error object itself